src/inventory.rs

```diff
@@ -126,6 +126,21 @@ impl<C: Config> Inventory<C> {
         mem::replace(old, new)
     }
 
+    pub fn swap_slot(&mut self, idx_a: u16, idx_b: u16) {
+        assert!(idx_a < self.slot_count(), "slot index out of range");
+        assert!(idx_b < self.slot_count(), "slot index out of range");
+
+        if idx_a == idx_b {
```

Comment:
We can avoid marking the slots as modified and swapping if the contents of the slots at the indices are equal.

Reply:
Yes, that is what this does.

Reply:
I meant comparing the items themselves, not just the indices.

Reply:
oh the **contents** lmao. yeah I'll make that change